peakinvestigator/lib/progress2.py

```diff
@@ -0,0 +1,37 @@
+## -*- coding: utf-8 -*-
+#
+# Copyright (c) 2016, Veritomyx, Inc.
+#
+# This file is part of the Python SDK for PeakInvestigator
+# (http://veritomyx.com) and is distributed under the terms
+# of the BSD 3-Clause license.
+import logging
+import time
+import sys
+import os
+
+logging.getLogger('Progress')
+
+class Progress_Factory2(object):
+
+    def __init__(self, total, unit='scans'):
+        self.total = total
+        self.unit = unit
+        self.count = 0
+        self.clock = time.time()
+        self.logger = logging.getLogger('Progress')
+        self.logger.setLevel(logging.INFO)
+
+    def update(self, update):
+        self.count += update
+        current = time.time()
+        if round(self.count/self.total*100,0) % 10 == 0:
+            sys.stdout.write('\r'+str(round(self.count/self.total*100,0))+str(' Percent'))
```

Comment:
Any reason you're using `sys.stdout` instead of `self.logger` or even `print`?

Reply:
Print appends strings, where sys.stdout.write enables it to be written over (instead of have multiple lines showing 
10
20
30
40, only one line is used.

Reply:
self.logger in my initial test had the same issue as print.

Reply:
Ah, clever! Forgot about those console tricks.